networking/eth2/src/test/java/tech/pegasys/artemis/networking/eth2/gossip/AttestationTopicSubscriptionsTest.java

```diff
@@ -82,12 +85,67 @@ public void shouldPreserveLaterSubscriptionPeriodWhenEarlierSlotAdded() {
     final UnsignedLong secondSlot = UnsignedLong.valueOf(15);
 
     subscriptions.subscribeToCommittee(committeeIndex, secondSlot);
-    subscriptions.subscribeToCommittee(committeeIndex, firstSlot);
 
     subscriptions.onSlot(firstSlot.plus(ONE));
     verify(eth2Network, never()).unsubscribeFromAttestationCommitteeTopic(committeeIndex);
 
     subscriptions.onSlot(secondSlot.plus(ONE));
     verify(eth2Network).unsubscribeFromAttestationCommitteeTopic(committeeIndex);
   }
+
+  @Test
+  public void shouldSubscribeToSubnet() {
+    final int committeeIndex = 3;
+
+    final UnsignedLong firstSlot = UnsignedLong.valueOf(10);
+    subscriptions.subscribeToCommittee(committeeIndex, firstSlot);
+    int subnetIndex = committeeIndex % ATTESTATION_SUBNET_COUNT;
+    Bitvector bitvector = new Bitvector(ATTESTATION_SUBNET_COUNT);
+    bitvector.setBit(subnetIndex);
+
+    verify(eth2Network).updateAttestationSubnetENRField(bitvector.serialize());
+  }
+
+  @Test
+  public void shouldSubscribeToMultipleSubnets() {
+    final int committeeIndex1 = 3;
+    final int committeeIndex2 = 10;
+
+    final UnsignedLong firstSlot = UnsignedLong.valueOf(10);
+    subscriptions.subscribeToCommittee(committeeIndex1, firstSlot);
+    subscriptions.subscribeToCommittee(committeeIndex2, firstSlot);
+    int subnetIndex1 = committeeIndex1 % ATTESTATION_SUBNET_COUNT;
+    int subnetIndex2 = committeeIndex2 % ATTESTATION_SUBNET_COUNT;
+    Bitvector bitvector = new Bitvector(ATTESTATION_SUBNET_COUNT);
+
+    bitvector.setBit(subnetIndex1);
+    verify(eth2Network).updateAttestationSubnetENRField(bitvector.serialize());
+
+    bitvector.setBit(subnetIndex2);
+    verify(eth2Network).updateAttestationSubnetENRField(bitvector.serialize());
+  }
+
+  @Test
+  public void shouldNotUnsubscribeWhenThereAreOtherCommitteeThatMapToSameSubnet() {
+
+    final int committeeIndex1 = 4;
+    final int committeeIndex2 = 4 + ATTESTATION_SUBNET_COUNT;
+
+    final UnsignedLong slot1 = UnsignedLong.valueOf(10);
+    final UnsignedLong slot2 = UnsignedLong.valueOf(13);
+    subscriptions.subscribeToCommittee(committeeIndex1, slot1);
+    subscriptions.subscribeToCommittee(committeeIndex2, slot2);
+
+    Bitvector bitvector = new Bitvector(ATTESTATION_SUBNET_COUNT);
+    bitvector.setBit(4);
+
+    verify(eth2Network, times(1)).updateAttestationSubnetENRField(bitvector.serialize());
+
+    subscriptions.onSlot(slot1.plus(ONE));
+    verify(eth2Network, times(1)).updateAttestationSubnetENRField(bitvector.serialize());
```

Comment:
I suspect this should be
```suggestion
    verify(eth2Network, times(1)).updateAttestationSubnetENRField(any());
```

Otherwise the test will pass if the ENR field is updated with a different value because it's only asserting that particular value is only set once.

Reply:
Yeah. Good catch.